lxd/instance/drivers/driver_qemu.go

```diff
@@ -5284,7 +5298,7 @@ func (d *qemu) Update(args db.InstanceArgs, userRequested bool) error {
 
 	if d.architectureSupportsUEFI(d.architecture) && (shared.ValueInSlice("security.secureboot", changedConfig) || shared.ValueInSlice("security.csm", changedConfig)) {
 		// Re-generate the NVRAM.
-		err = d.setupNvram()
+		err = d.setupNvram(true)
```

Comment:
I would prefer that setupNvram does not perform any mounts that instead we ensure that the root drive is mounted before calling it, where appropriate/needed.

Reply:
Have done. Unfortunately I wasn't able to find any clear and consistent way to check if config volume is mounted, most likely we need to extend our storage drivers API and add something like `d.isMounted(...)` or something like that. For now, I have just factored mounting out from the `setupNvram` and added an explicit `d.mount` call where it's needed.